stories/2024-01-01-Eurovision/story.ipynb

```diff
@@ -2840,6 +2840,674 @@
+                "    Y = Y.astype(int)\n",
+                "\n",
+                "    X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=test_size, random_state=seed, stratify=X['points'])\n",
+                "\n",
+                "    test_all = X_test\n",
+                "\n",
+                "    # remove 'Votes' and 'received_vote' columns from X_train\n",
+                "    X_train = X_train[X_train.columns.drop('points')]\n",
+                "    X_train = X_train[X_train.columns.drop('received_vote')]\n",
+                "    X_test = X_test[X_test.columns.drop('points')]\n",
+                "    X_test = X_test[X_test.columns.drop('received_vote')]\n",
+                "\n",
+                "    model = xgb.XGBClassifier()\n",
+                "    model.fit(X_train, y_train)\n",
+                "\n",
+                "    return model, X_test, y_test, test_all\n",
+                "\n",
+                "def model_predictions(model, X_test, y_test):\n",
+                "    Xt = X_test.copy()\n",
+                "    y_pred = model.predict(Xt)\n",
+                "    predictions = [round(value) for value in y_pred]\n",
+                "\n",
+                "    Xt['prob'] = model.predict_proba(Xt)[:,1]\n",
+                "    Xt['predictions'] = predictions\n",
+                "    Xt['actual'] = y_test\n",
+                "    return Xt\n",
+                "\n",
+                "def model_evalutation(df_pred):\n",
+                "    # evaluate predictions\n",
+                "    accuracy = accuracy_score(df_pred['actual'], df_pred['predictions'])\n",
+                "    print(\"Accuracy: %.2f%%\" % (accuracy * 100.0))\n",
+                "\n",
+                "    # print a confusion matrix heatmap\n",
+                "    cm = confusion_matrix(df_pred['actual'], df_pred['predictions'])\n",
+                "    sns.heatmap(cm, annot=True, fmt=\"d\", cmap=\"Blues\")\n",
+                "\n",
+                "    # add title and axis labels\n",
+                "    plt.title('Confusion matrix' + \"(Accuracy={:.3f})\".format(accuracy))\n",
+                "    plt.ylabel('True label')\n",
+                "    plt.xlabel('Predicted label')"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "model_basic, X_test, y_test, test_all = basic_xgboost(df_xgboost, seed=7, test_size=0.33)\n",
+                "predictions = model_predictions(model_basic, X_test, y_test)\n",
+                "model_evalutation(predictions)"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "graph = xgb.to_graphviz(model_basic, num_trees=1, rankdir='LR')\n",
```

Comment:
this is very nice, but I had to `brew install graphviz` to make this work (which I think is fine, but might be worth a comment)

Reply:
just to note: I also did `pip install graphviz` before, but that wasn't sufficient it seems (but maybe we should add it to the requirements anyway)